### DISCUSSION TITLE
How to rotate a fragment around rotatable bond

### DISCUSSION THREAD

**Discussion by TAEUKHAM** [rdkit/rdkit#4908] [Q&A] [answered]
Hi! I'm trying to rotate a fragment around a rotatable bond of a molecule like this.  

![20220119_164819](https://user-images.githubusercontent.com/88373558/150086796-afb108dc-e651-4bfd-b048-6a064ccd5702.png)
![20220119_164833](https://user-images.githubusercontent.com/88373558/150086825-358e2bfb-8816-42d2-9a24-0a63c98fd08a.png)

I've found somebody had [questioned this](https://sourceforge.net/p/rdkit/mailman/rdkit-discuss/thread/CAD4fdRRsYVw5tGvQWoHmZm%3DMgJVAD7oEsXrsSK6kfL%3DS41Ajgg%40mail.gmail.com/#msg31413606) in 2013 and got answered back that it was not implemented in python library(yet). I wonder if it is possible now:D For now, I'm doing this by directly modifying the coordinates of molfiles by finding each fragment's atom index and bond index. 

Thanks!

**Comment by ptosco** [comment, marked as answer]
@TAEUKHAM You can select the dihedral angle corresponding to the bond you wish to rotate around using a 4-tuple of atom indices and then use [rdMolTransforms.SetDihedralDeg](http://rdkit.org/docs/source/rdkit.Chem.rdMolTransforms.html?highlight=setdihedral#rdkit.Chem.rdMolTransforms.SetDihedralDeg) or [rdMolTransforms.SetDihedralRad](http://rdkit.org/docs/source/rdkit.Chem.rdMolTransforms.html?highlight=setdihedral#rdkit.Chem.rdMolTransforms.SetDihedralRad) (using respectively degrees or radians as measure unit) to set the dihedral angle.

**Comment by TAEUKHAM** [comment]
@ptosco Thanks ptosco! You are my lifesaver:D
